server/service/src/processors/general_processor.rs

```diff
@@ -120,15 +131,16 @@ pub(super) trait Processor {
 
     fn cursor_type(&self) -> KeyType;
 
-    fn try_process_record_common(
+    async fn try_process_record_common(
         &self,
         ctx: &ServiceContext,
         service_provider: &ServiceProvider,
         changelog: &ChangelogRow,
     ) -> Result<Option<String>, ProcessorError> {
-        let result = ctx
-            .connection
-            .transaction_sync(|_| self.try_process_record(ctx, service_provider, changelog))
+        // TODO: should be in a transaction, we need to support transaction_async
+        let result = self
+            .try_process_record(ctx, service_provider, changelog)
+            .await
```

Comment:
had to remove transaction to make async, but I checked, all processors do max of 1 DB write for now, so I think ok, I can make an issue?